app/battle/battle.py

```diff
@@ -0,0 +1,123 @@
+def battle(knightsConfig: dict):
+    # BATTLE PREPARATIONS:
+
+    # lancelot
+    lancelot = knightsConfig["lancelot"]
+
+    # apply armour
+    lancelot["protection"] = 0
+    for a in lancelot["armour"]:
+        lancelot["protection"] += a["protection"]
+
+    # apply weapon
+    lancelot["power"] += lancelot["weapon"]["power"]
+
+    # apply potion if exist
+    if lancelot["potion"] is not None:
+        if "power" in lancelot["potion"]["effect"]:
+            lancelot["power"] += lancelot["potion"]["effect"]["power"]
+
+        if "protection" in lancelot["potion"]["effect"]:
+            lancelot["protection"] += lancelot["potion"]["effect"]["protection"]
+
+        if "hp" in lancelot["potion"]["effect"]:
+            lancelot["hp"] += lancelot["potion"]["effect"]["hp"]
+
+    # arthur
+    arthur = knightsConfig["arthur"]
+
+    # apply armour
+    arthur["protection"] = 0
+    for a in arthur["armour"]:
+        arthur["protection"] += a["protection"]
+
+    # apply weapon
+    arthur["power"] += arthur["weapon"]["power"]

```

Comment:
Similar to the issue with Lancelot, ensure that `arthur['power']` is initialized before this line to avoid a `KeyError`.